src/ansi-c/c_typecheck_base.cpp

```diff
@@ -645,20 +641,21 @@ void c_typecheck_baset::typecheck_declaration(
     irept contract;
 
     {
-      exprt spec_requires=
-        static_cast<const exprt&>(declaration.find(ID_C_spec_requires));
-      contract.add(ID_C_spec_requires).swap(spec_requires);
+      exprt spec_assigns = declaration.spec_assigns();
```

Comment:
While this is a clear improvement, it is one of those changes that could easily be placed in a commit of its own. This would make reviewing much much easier.

Reply:
I added it to a separate commit here.